t.go

```diff
@@ -203,3 +219,263 @@ func (t *Torrent) AddTrackers(announceList [][]string) {
 	defer t.cl.mu.Unlock()
 	t.addTrackers(announceList)
 }
+
+func (t *Torrent) RemoveTracker(addr string) {
+	nn := make([][]string, 0)
+
+	for _, tier := range t.metainfo.AnnounceList {
+		n := make([]string, 0)
+		for _, tracker := range tier {
+			if tracker != addr {
+				n = append(n, tracker)
+			} else {
+				if trackerScraper, ok := t.trackerAnnouncers[tracker]; ok {
+					trackerScraper.stop.Set()
+				}
+			}
+		}
+		if len(n) > 0 {
+			nn = append(nn, n)
+		}
+	}
+
+	t.metainfo.AnnounceList = nn
+}
+
+type Tracker struct {
+	Url          string
+	Peers        int
+	Err          string
+	LastAnnounce int64
+	NextAnnounce int64
+}
+
+func (t *Torrent) Trackers() []Tracker {
+	t.cl.mu.Lock()
+	defer t.cl.mu.Unlock()
+
+	var tt []Tracker
+
+	for _, tier := range t.announceList() {
+		for _, trackerURL := range tier {
+			T := Tracker{Url: trackerURL}
+			if trackerScraper, ok := t.trackerAnnouncers[trackerURL]; ok {
+				T.Peers = trackerScraper.peers
+				T.Err = trackerScraper.err
+				T.LastAnnounce = trackerScraper.lastAnnounce
+				T.NextAnnounce = trackerScraper.nextAnnounce
+			}
+			tt = append(tt, T)
+		}
+	}
+
+	tt = append(tt, Tracker{"DHT", t.peersDHT, t.errDHT, t.lastAnnounceDHT, 0})
+
+	tt = append(tt, Tracker{"PEX", t.peersPEX, "", 0, 0})
+
+	return tt
+}
+
+type PeerInfo struct {
+	Id     [20]byte
+	Name   string
+	Addr   string
+	Source peerSource
+	// Peer is known to support encryption.
+	SupportsEncryption bool
+	PiecesCompleted    int
+	// byte info information
+	Downloaded int64
+	Uploaded   int64
+}
+
+func (t *Torrent) PeersCount() int {
+	t.cl.mu.Lock()
+	defer t.cl.mu.Unlock()
+	return len(t.peers)
+}
+
+func (t *Torrent) Peers() []PeerInfo {
+	t.cl.mu.Lock()
+	defer t.cl.mu.Unlock()
+
+	v := make([]PeerInfo, 0, len(t.conns))
+	for _, value := range t.conns {
+		v = append(v, PeerInfo{value.PeerID, value.PeerClientName,
+			value.remoteAddr().String(), value.Discovery, value.encrypted,
+			value.peerCompleted, value.Downloaded, value.Uploaded})
+	}
+	return v
+}
+
+func (t *Torrent) PiecePended(i int) bool {
+	t.cl.mu.Lock()
+	defer t.cl.mu.Unlock()
+
+	return t.pendingPieces.Contains(i)
+}
```

Comment:
I think there's already a piece state method that should be augmented with this, rather than adding a separate function.